### ISSUE TITLE
Support for accessing HTML template's holes dynamically

### ISSUE THREAD

**Issue by ivaylo5ev** [dotnet-websharper/ui#201]
As per [WebSharper 3 documentation examples](https://developers.websharper.com/docs/v3.x/fs/html-templates) one could access holes dynamically by passing the hole name as a string:

**HTML Template**
```
<article>
  <h2>${title}</h2>
  <section class="main" data-hole="content"></section>
</article>
```

**F# Code**
```
type Article =
    {
        Title : string
        Content : list<Element>
    }

let ArticleTemplate =
        Content.Template<Article>("~/Article.html")
            .With("title", fun x -> x.Title)
            .With("content", fun x -> x.Content)
```
I cannot seem to find a way to do this with the new type provider. Currently I face a scenario, as described in [my stackoverflow post](https://stackoverflow.com/questions/52984353/can-i-fill-a-websharper-templates-hole-without-knowing-its-name-at-compile-time), where I need to populate a content hole without knowing its name at compile time. Unless I am missing something, I feel the current implementation does not cover my case.

Would it be possible to add the ability to access holes dynamically even from type-provider-managed template instances?

**Comment by Tarmil** [comment]
Yes, this makes sense and should be useful.

I have a preliminary implementation of `.With(name, value)` for the type provider that I'll push soon. We may also want to have a type that is not provided, so that you can load fully dynamic templates; although it should basically be equivalent to `Template<"<div>">`.

**Comment by ivaylo5ev** [comment]
This is great! I am looking forward for this to get merged. Thanks1

**Comment by Tarmil** [comment]
This is now implemented (for the server side) and released as v4.5.5. I will open a separate issue for the client side version.
